omod-dstu3/src/main/java/org/openmrs/module/fhir2/providers/PatientDstu3FhirResourceProvider.java

```diff
@@ -0,0 +1,87 @@
+
+import ca.uhn.fhir.rest.annotation.*;
+import ca.uhn.fhir.rest.api.SortSpec;
+import ca.uhn.fhir.rest.param.DateRangeParam;
+import ca.uhn.fhir.rest.param.StringOrListParam;
+import ca.uhn.fhir.rest.param.TokenOrListParam;
+import ca.uhn.fhir.rest.server.IResourceProvider;
+import ca.uhn.fhir.rest.server.exceptions.ResourceNotFoundException;
+import lombok.AccessLevel;
+import lombok.Setter;
+import org.hl7.fhir.convertors.VersionConvertor_30_40;
+import org.hl7.fhir.dstu3.model.*;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.openmrs.module.fhir2.api.FhirPatientService;
+import org.openmrs.module.fhir2.util.FhirServerUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+@Component
+@Qualifier("fhirDstu3Resources")
+@Setter(AccessLevel.PACKAGE)
+public class PatientDstu3FhirResourceProvider implements IResourceProvider {
+	
+	@Autowired
+	private FhirPatientService patientService;
+	
+	@Override
+	public Class<? extends IBaseResource> getResourceType() {
+		return Patient.class;
+	}
+	
+	@Read
+	@SuppressWarnings("unused")
+	public Patient getPatientById(@IdParam @NotNull IdType id) {
+		org.hl7.fhir.r4.model.Patient patient = patientService.getPatientByUuid(id.getIdPart());
+		if (patient == null) {
+			throw new ResourceNotFoundException("Could not find patient with Id " + id.getIdPart());
+		}
+		
+		return VersionConvertor_30_40.convertPatient(patient);
+	}
+	
+	@Search
+	@SuppressWarnings("unused")
+	public Bundle searchPatients(@OptionalParam(name = Patient.SP_NAME) StringOrListParam name,
+	        @OptionalParam(name = Patient.SP_GIVEN) StringOrListParam given,
+	        @OptionalParam(name = Patient.SP_FAMILY) StringOrListParam family,
+	        @OptionalParam(name = Patient.SP_IDENTIFIER) TokenOrListParam identifier,
+	        @OptionalParam(name = Patient.SP_GENDER) TokenOrListParam gender,
+	        @OptionalParam(name = Patient.SP_BIRTHDATE) DateRangeParam birthDate,
+	        @OptionalParam(name = Patient.SP_DEATH_DATE) DateRangeParam deathDate,
+	        @OptionalParam(name = Patient.SP_DECEASED) TokenOrListParam deceased,
+	        @OptionalParam(name = Patient.SP_ADDRESS_CITY) StringOrListParam city,
+	        @OptionalParam(name = Patient.SP_ADDRESS_STATE) StringOrListParam state,
+	        @OptionalParam(name = Patient.SP_ADDRESS_POSTALCODE) StringOrListParam postalCode,
+	        @OptionalParam(name = Patient.SP_ADDRESS_COUNTRY) StringOrListParam country, @Sort SortSpec sort) {
+		return VersionConvertor_30_40
+		        .convertBundle(FhirServerUtils.convertSearchResultsToBundle(patientService.searchForPatients(name, given,
+		            family, identifier, gender, birthDate, deathDate, deceased, city, state, postalCode, country, sort)));
+	}
+	
+	@History
+	@SuppressWarnings("unused")
+	public List<Resource> getPatientResourceHistory(@IdParam @NotNull IdType id) {
```

Comment:
`getPatientHistoryById`, also ordering.

Reply:
@ibacher This might need to be changed in master as well

Reply:
I'm aware. I've got to go through that code soon.